test/jdk/java/awt/print/PrinterJob/raster/RasterTest.java

```diff
@@ -21,248 +21,143 @@
-
-     }
-
-    public Dimension getPreferredSize() {
-        return new Dimension(500, 500);
-    }
+    public void actionPerformed(ActionEvent e) {
 
- }
+        PrinterJob pj = PrinterJob.getPrinterJob();
 
-}
-
-
-class Sysout {
-   private static TestDialog dialog;
-
-   public static void createDialogWithInstructions( String[] instructions )
-    {
-      dialog = new TestDialog( new Frame(), "Instructions" );
-      dialog.printInstructions( instructions );
-      dialog.show();
-      println( "Any messages for the tester will display here." );
-    }
-
-   public static void createDialog( )
-    {
-      dialog = new TestDialog( new Frame(), "Instructions" );
-      String[] defInstr = { "Instructions will appear here. ", "" } ;
-      dialog.printInstructions( defInstr );
-      dialog.show();
-      println( "Any messages for the tester will display here." );
-    }
-
-
-   public static void printInstructions( String[] instructions )
-    {
-      dialog.printInstructions( instructions );
+        if (pj != null && pj.printDialog()) {
+            pj.setPrintable(c);
+            try {
+                pj.print();
+            } catch (PrinterException pe) {
+                pe.printStackTrace();
+            } finally {
+                System.out.println("PRINT RETURNED");
+            }
+        }
     }
 
-
-   public static void println( String messageIn )
-    {
-      dialog.displayMessage( messageIn );
+    static class RasterCanvas extends Canvas implements Printable {
+
+        public int print(Graphics g, PageFormat pgFmt, int pgIndex) {
+            if (pgIndex > 0)
+                return Printable.NO_SUCH_PAGE;
+
+            Graphics2D g2d = (Graphics2D) g;
+            g2d.translate(pgFmt.getImageableX(), pgFmt.getImageableY());
+            doPaint(g2d);
+            return Printable.PAGE_EXISTS;
+        }
+
+        public void paint(Graphics g) {
+            doPaint(g);
+        }
+
+        public void paintComponent(Graphics g) {
+            doPaint(g);
+        }
+
+        public void doPaint(Graphics g) {
+            Graphics2D g2 = (Graphics2D) g;
+
+            g2.setColor(Color.black);
+
+            BufferedImage bimg = new BufferedImage(200, 200,
+                    BufferedImage.TYPE_INT_ARGB);
+            Graphics ig = bimg.getGraphics();
+            Color alphared = new Color(255, 0, 0, 128);
+            Color alphagreen = new Color(0, 255, 0, 128);
+            Color alphablue = new Color(0, 0, 255, 128);
+            ig.setColor(alphared);
+            ig.fillRect(0, 0, 200, 200);
+            ig.setColor(alphagreen);
+            ig.fillRect(25, 25, 150, 150);
+            ig.setColor(alphablue);
+            ig.fillRect(75, 75, 125, 125);
+            g.drawImage(bimg, 10, 25, this);
+
+            GradientPaint gp =
+                    new GradientPaint(10.0f, 10.0f, alphablue, 210.0f, 210.0f, alphared, true);
+            g2.setPaint(gp);
+            g2.fillRect(10, 240, 200, 200);
```

Comment:
dispose g2D object (and graphics object)

Reply:
Do not dispose of the `Graphics` object you didn't create.